core/trino-main/src/main/java/io/trino/operator/SimplePagesHashStrategy.java

```diff
@@ -279,11 +280,12 @@ public boolean isPositionNull(int blockIndex, int blockPosition)
     public int compareSortChannelPositions(int leftBlockIndex, int leftBlockPosition, int rightBlockIndex, int rightBlockPosition)
     {
         int channel = getSortChannel();
+        checkState(comparisonOperators.get(channel).isPresent(), "type is not orderable");
```

Comment:
Do we need to check the state for each invocation ?

Reply:
@dain Can `SimplePagesHashStrategy` have more than one sort channel ?

Reply:
> Do we need to check the state for each invocation ?

It would be possible to check in the conductor and preserve the check result as a boolean field. Even in that case, checking the boolean field would be necessary here.